src/main/java/org/symphonyoss/symphony/messageml/elements/TableBody.java

```diff
@@ -43,4 +44,26 @@ public void validate() throws InvalidInputException {
     assertNoText();
     assertContentModel(Collections.<Class<? extends Element>>singleton(TableRow.class));
   }
-}
+
+
+  @Override
+  public void asPresentationML(XmlPrintStream out) {
+    Element parent = getParent();
+
+    if (!TableSelect.class.equals(parent.getClass())) {
+      super.asPresentationML(out);
+      return;
+    }
+
+    Long rowNumber = 1L;
+    for (Element child : getChildren()) {
```

Comment:
This shouldn't be on this method in my pov, the Row should be initialized already with the rowNumber instead of the parent setting it for the row.